libs/traits/src/lib.rs

```diff
@@ -281,56 +285,103 @@ pub trait CurrencyPrice<CurrencyId> {
 	) -> Option<PriceValue<CurrencyId, Self::Rate, Self::Moment>>;
 }
 
+#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+pub enum CompoundingSchedule {
+	/// Interest compounds every second
+	Secondly,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+pub enum InterestRate<Rate> {
+	/// Interest accrues at a fixed rate
+	Fixed {
+		rate_per_year: Rate,
+		compounding: CompoundingSchedule,
+	},
+}
+
+impl<Rate: EnsureAdd + EnsureSub> InterestRate<Rate> {
+	pub fn ensure_add(self, rate: Rate) -> Result<InterestRate<Rate>, ArithmeticError> {
+		match self {
+			InterestRate::Fixed {
+				rate_per_year,
+				compounding,
+			} => {
+				let new_rate = rate_per_year.ensure_add(rate)?;
+				Ok(InterestRate::Fixed {
+					rate_per_year: new_rate,
+					compounding,
+				})
+			}
+		}
+	}
+
+	pub fn ensure_sub(self, rate: Rate) -> Result<InterestRate<Rate>, ArithmeticError> {
+		match self {
+			InterestRate::Fixed {
+				rate_per_year,
+				compounding,
+			} => {
+				let new_rate = rate_per_year.ensure_sub(rate)?;
+				Ok(InterestRate::Fixed {
+					rate_per_year: new_rate,
+					compounding,
+				})
+			}
+		}
+	}
+}
```

Comment:
I think could be great to add a `map/try_map` function to `InterestRate` and let the caller call the operation they want over the rate. Similar to https://github.com/centrifuge/centrifuge-chain/pull/1435/files#diff-14aad6e50ab4015efc0d5939359c1676112c04751b9a825c15e1a588677da0b6R42

Reply:
I actually prefer ensure add impl, because this is slightly more specific. The advantage is, say at some point there is also a floating rate variant next to the fixed variant. Then with map, it's unclear what field this should touch, and the implementor would need to handle both cases. Versus with ensure add the impl on interest rate can decide that adding always means increasing the inner rate.

Reply:
But we can still definitely also add `map/try_map` in the future, it can't hurt

Reply:
`map/try_map` usually only acts over the most important property. i.e. `map` only maps the ok branch in a `Result` type. For this type, the core field is the inner rate, so I think it makes sense to only map over it. Other properties are secondary. 

Anyways, it can makes sense to have an `ensure_add()` method directly in `InterestRate`. In that case, it can be built using `try_map` to factorize most of the code as follows:

```rust
fn ensure_add(&self, rate: Rate) -> Result<Self, DispatchError> {
    self.try_map(|rate_per_year| rate_per_year.ensure_add(rate))
}

fn ensure_sub(&self, rate: Rate) -> Result<Self, DispatchError> {
    self.try_map(|rate_per_year| rate_per_year.ensure_sub(rate))
}
```

Reply:
Of course, any of this is required for this PR, and could be added in the future 👍🏻